johnzon-jsonb/src/main/java/org/apache/johnzon/jsonb/JsonbAccessMode.java

```diff
@@ -585,6 +585,8 @@ private boolean isReversedAdapter(final Class<?> payloadType, final Class<?> aCl
         }
         final Type[] genericInterfaces = aClass.getGenericInterfaces();
         return Stream.of(genericInterfaces).filter(ParameterizedType.class::isInstance)
+                .filter(i -> ParameterizedType.class.cast(i).getRawType() instanceof Class)
+                .filter(i -> ParameterizedType.class.cast(i).getActualTypeArguments()[0] instanceof Class)
```

Comment:
maybe check the length too in this lambda, can fail otherwise ;)

Reply:
done